server/src/main/java/org/opensearch/index/mapper/FlatObjectFieldMapper.java

```diff
@@ -0,0 +1,745 @@
+                        + value
+                        + "\""
+                );
+            }
+            ts.end();
+            return newValue;
+        }
+    }
+
+    @Override
+    protected String contentType() {
+        return CONTENT_TYPE;
+    }
+
+    private static final class ValueAndPathFieldMapper extends FieldMapper {
+
+        protected ValueAndPathFieldMapper(FieldType fieldType, KeywordFieldMapper.KeywordFieldType mappedFieldType) {
+            super(mappedFieldType.name(), fieldType, mappedFieldType, MultiFields.empty(), CopyTo.empty());
+        }
+
+        void addField(ParseContext context, String value) {
+            final BytesRef binaryValue = new BytesRef(value);
+            if (fieldType.indexOptions() != IndexOptions.NONE || fieldType.stored()) {
+                Field field = new KeywordFieldMapper.KeywordField(fieldType().name(), binaryValue, fieldType);
+
+                context.doc().add(field);
+
+                if (fieldType().hasDocValues() == false && fieldType.omitNorms()) {
+                    createFieldNamesField(context);
+                }
+            }
+        }
+
+        @Override
+        protected void parseCreateField(ParseContext context) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        protected void mergeOptions(FieldMapper other, List<String> conflicts) {
+
+        }
+
+        @Override
+        protected String contentType() {
+            return "valueAndPath";
+        }
+
+        @Override
+        public String toString() {
+            return fieldType().toString();
+        }
+
+    }
+
+    private static final class ValueFieldMapper extends FieldMapper {
+
+        protected ValueFieldMapper(FieldType fieldType, KeywordFieldMapper.KeywordFieldType mappedFieldType) {
+            super(mappedFieldType.name(), fieldType, mappedFieldType, MultiFields.empty(), CopyTo.empty());
+        }
+
+        void addField(ParseContext context, String value) {
+            final BytesRef binaryValue = new BytesRef(value);
+            if (fieldType.indexOptions() != IndexOptions.NONE || fieldType.stored()) {
+                Field field = new KeywordFieldMapper.KeywordField(fieldType().name(), binaryValue, fieldType);
+                context.doc().add(field);
+
+                if (fieldType().hasDocValues() == false && fieldType.omitNorms()) {
+                    createFieldNamesField(context);
+                }
+            }
+        }
+
+        @Override
+        protected void parseCreateField(ParseContext context) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        protected void mergeOptions(FieldMapper other, List<String> conflicts) {
+
+        }
+
+        @Override
+        protected String contentType() {
+            return "value";
+        }
+
+        @Override
+        public String toString() {
+            return fieldType().toString();
+        }
+    }
```

Comment:
`ValueAndPathFieldMapper` and `ValueFieldMapper` has no diff besides `contentType`, we should consider refactor it.

Reply:
The intention of making two FieldMapper classes is to allow different parameters settings in the future. Currently, we applied the default setting and do not open parameters yet.